src/sentry/search/snuba/executors.py

```diff
@@ -1564,6 +1580,16 @@ def query(
                 Condition(Column("timestamp", joined_entity), Op.LT, end),
             ]
             having = []
+            # if we need to prefetch from postgres, we add filter by the group ids
+            if group_ids_to_pass_to_snuba is not None:
+                # will not find any matches, we can return early
+                if len(group_ids_to_pass_to_snuba) == 0:
+                    return self.empty_result
+
+                where_conditions.append(
+                    Condition(Column("group_id", attr_entity), Op.IN, group_ids_to_pass_to_snuba)
+                )
```

Comment:
I wonder if we should filter on both the `event` and `attr` group id? I guess the join is doing that already, so maybe it doesn't matter...

Reply:
@wedamija not a bad idea, worth adding